src/cryptoadvance/specter/templates/includes/recipient-box.html

```diff
@@ -0,0 +1,486 @@
+						<qr-scanner id="address-scan" style="margin-top: 3px;">
+							<a slot="button" class="btn" style="height: 35px;">
+							<img src="{{ url_for('static', filename='img/qr-code.svg') }}" style="width: 26px; margin: 0px;" class="svg-white"> {{ _("Scan") }}</a>
+						</qr-scanner>
+					</div>				
+					<input type="text" id="label" class="address-label" name="label" value="" placeholder='{{ _("Address label") }}'>
+					<br>
+					{{ _("Amount:") }}<br>
+					<input style="width: 200px" type="number" name="amount" id="amount" min=0 step='1e-8' autocomplete="off" placeholder="0"> <!-- the value comes in here-->
+					<div class="mobile-only" style="margin-top: 10px;"></div>
+					<span id='asset_selector'><!-- the assetSelector comes in here--></span>
+					<span class="note max-btn" style="margin-left: 5px;" id="send_max">({{ _("send max") }})</span>
+					<div>
+						<span class="note" id="converted_unit_amount">-</span> <span class="note" id="converted_unit_label"><!-- the convertedUnit comes in here--></span> <span class="note" id="converted_unit_alt"></span>
+					</div>
+				</div>
+			<div>
+</template>
+
+
+
+<script type="text/javascript">
+	class RecipientBox extends HTMLElement {
+		constructor() {
+			super();
+			
+			this.internals = this.attachInternals();
+			this.recipientId = null;	// MUST be set for a functioning instance
+			this.inputs = {};   		// this is a dictionary containing all inputs, such they are easily accessible 
+			this.btcAmount = null; 	// will be calculated from the inputs
+			this.unit =  null;			// will be calculated from the inputs
+
+			this.buildHTML();			// created the html
+			this.attachListeners();		// attaches the listeners to some html objects
+		}
+
+
+
+		// creates the entire html code (without event listeners)  mainly from the template
+		// and it also populates the dictionary this.inputs
+		buildHTML(){
+			// Create a shadow root
+			this.attachShadow({
+				mode: 'open'
+			});
+
+			var template_content = document.getElementById('recipient-box').content;
+            this.clone = template_content.cloneNode(true);
+
+
+			// fill this.inputs with the input and textarea fields. 
+			// This dict is a quick way to look up the input fields without querying the html again
+			this.inputs = {};
+			var formClassNames = ['input', 'textarea'];
+			for (var j in formClassNames) {
+				var inputFields = this.clone.querySelectorAll(formClassNames[j]);
+				for(var i = 0; i < inputFields.length; i++) {
+					var inputField = inputFields[i];
+					console.log(j, i, inputField)
+					this.inputs[inputField.id] = inputField;
+				}
+			}
+
+
+            this.shadowRoot.appendChild(this.clone);
+			this.addAssetSelectorToShadowRoot('', 'btc')			
+		}
+
+
+
+		// add the html code for the assetSelector
+		addAssetSelectorToShadowRoot(amount, amountUnit) { 
+			if (amount >= 0){
+				this.shadowRoot.getElementById('amount').value = amount;
```

Comment:
This is due to my lack of understanding why we need clones.   AFAIK a ShadowRoot is simply a sub-html-structure, in which the elements (and their ids) are nicely encapsulated. That way I can choose simple ids like `id="remove"` and don't have to do `id="remove_0"`.  
Why would we need clones?

Reply:
I mean you also did it like this here: https://kinsta.com/blog/web-components/#html-templates
The only difference is how to query HMTL elements when assigning them as properties of the HTML element. I think in your approach you were fine with using the Shadow Root since you had already appended the clone of the template to the Shadow Root and in other web components in the code base the querying had to be done through the clone of the template because the Shadow Root had not been appended yet. 

This is a good and simple example from `dnd-textarea.html` (querying takes place before the `appendChild(clone)`):

```
constructor() {
  let shadow = this.attachShadow({mode: 'open'});
  let template = document.getElementById('dnd-textarea');
  let clone = template.content.cloneNode(true);
  this.textarea = clone.querySelector("textarea");
  shadow.appendChild(clone);
}
```

I suppose the main drawback is more lengthy code. Here for example you could have used `this.amount` instead of writing always `this.shadowRoot.getElementById('amount') `if you had defined` this.amount` in the constructor.

<img width="556" alt="grafik" src="https://user-images.githubusercontent.com/70536101/180608194-d4ab59c7-9442-4572-bebf-2d8938b24092.png">

We can leave it like this but we'd need at least an explanation / comment that this web component uses a little bit of different approach in this regard. Tbh, I can't say for sure what the best practice is.

Reply:
I would agree here to @moneymanolis. Javascript as such is already quite flexible and can be misused in a lot of ways. Having consistency in the project is a value in itself. 
I also see value in clear explicit logical (buiseness-) references: e.g. `amount` getter and setter.
However, referring to internal representation of these (business-) properties via some search-algorithm (like `shadowRoot.getElementById`) doesn't feel right. 
So ideally maybe we can have additional members in the class like `amountInput` or `amountElement` which are defined in the constructor and used instead of `shadowRoot.getElementById("amount")`?

Reply:
@k9ert , @moneymanolis  : Thanks for the advice. I followed `  this.textarea = clone.querySelector("textarea");`  in 5d3c188781c21d5afd35b686d02ad324cb4980ed  (which also removes old code).